includes/class-wcs-action-scheduler-customer-notifications.php

```diff
@@ -497,7 +497,25 @@ public static function get_valid_notifications( $subscription ) {
 			}
 		}
 
-		return $notifications;
+		/**
+		 * Filter: `woocommerce_subscription_valid_customer_notification_types`.
+		 *
+		 * Allows filtering the list of notification types that will be scheduled for a particular subscription.
+		 *
+		 * Default array format returned:
+		 *
+		 * array(
+		 *     'next_payment', // Exists if the subscription contains a next payment date in the future.
+		 *     'trial_end',    // Exists if the subscription contains a trial end date in the future.
+		 *     'end'           // Exists if the subscription contains an end date in the future.
+		 * )
+		 *
+		 * @since 7.2.0
+		 *
+		 * @param array           $notifications Array of valid notification types.
+		 * @param WC_Subscription $subscription  Subscription object.
+		 */
+		return apply_filters( 'woocommerce_subscription_valid_customer_notification_types', $notifications, $subscription );
```

Comment:
From a defensive coding perspective, what are your thoughts on applying a cast here?

```suggestion
		return (array) apply_filters( 'woocommerce_subscription_valid_customer_notification_types', $notifications, $subscription );
```

I'm thinking that, conceivably, the first callback might do something like `return false` (either inadvertently, or as a misconceived way of wiping the list of notifications). Then, if a hypothetical second callback did something like the following, we'd have a fatal error:

```php
function filter_callback_2( $notifications ) {
	$additional_notifications = [ 'foo', 'bar' ];
	return array_merge( $notifications, $additional_notifications ); # Oh-oh!
}
```

The cast doesn't solve all possible problems, but it might reduce the overall risk of fatals just a little.

Reply:
Thanks for making this suggestion!